apps/studio/components/interfaces/DiskManagement/DiskManagementForm.tsx

```diff
@@ -0,0 +1,442 @@
+import { PermissionAction } from '@supabase/shared-types/out/constants'
+import { useQueryClient } from '@tanstack/react-query'
+import { useParams } from 'common'
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import { MAX_WIDTH_CLASSES, PADDING_CLASSES, ScaffoldContainer } from 'components/layouts/Scaffold'
+import {
+  useDiskAttributesQuery,
+  useRemainingDurationForDiskAttributeUpdate,
+} from 'data/config/disk-attributes-query'
+import { useUpdateDiskAttributesMutation } from 'data/config/disk-attributes-update-mutation'
+import { useDiskUtilizationQuery } from 'data/config/disk-utilization-query'
+import { setProjectStatus } from 'data/projects/projects-query'
+import { useReadReplicasQuery } from 'data/read-replicas/replicas-query'
+import { useOrgSubscriptionQuery } from 'data/subscriptions/org-subscription-query'
+import { useProjectAddonUpdateMutation } from 'data/subscriptions/project-addon-update-mutation'
+import { useProjectAddonsQuery } from 'data/subscriptions/project-addons-query'
+import { AnimatePresence, motion } from 'framer-motion'
+import { useCheckPermissions, usePermissionsLoaded } from 'hooks/misc/useCheckPermissions'
+import { useSelectedOrganization } from 'hooks/misc/useSelectedOrganization'
+import { PROJECT_STATUS } from 'lib/constants'
+import { ChevronRight } from 'lucide-react'
+import { useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { toast } from 'sonner'
+import {
+  Button,
+  cn,
+  Collapsible_Shadcn_,
+  CollapsibleContent_Shadcn_,
+  CollapsibleTrigger_Shadcn_,
+  Form_Shadcn_,
+  Separator,
+} from 'ui'
+import { FormFooterChangeBadge } from '../DataWarehouse/FormFooterChangeBadge'
+import { CreateDiskStorageSchema, DiskStorageSchemaType } from './DiskManagement.schema'
+import { DiskMangementRestartRequiredSection } from './DiskManagementRestartRequiredSection'
+import { DiskManagementReviewAndSubmitDialog } from './DiskManagementReviewAndSubmitDialog'
+import { ComputeSizeField } from './fields/ComputeSizeField'
+import { DiskSizeField } from './fields/DiskSizeField'
+import { IOPSField } from './fields/IOPSField'
+import { StorageTypeField } from './fields/StorageTypeField'
+import { ThroughputField } from './fields/ThroughputField'
+import { DiskCountdownRadial } from './ui/DiskCountdownRadial'
+import {
+  DiskType,
+  IOPS_RANGE,
+  RESTRICTED_COMPUTE_FOR_THROUGHPUT_ON_GP3,
+} from './ui/DiskManagement.constants'
+import { SpendCapDisabledSection } from './ui/SpendCapDisabledSection'
+import { NoticeBar } from './ui/NoticeBar'
+import { mapComputeSizeNameToAddonVariantId, showMicroUpgrade } from './DiskManagement.utils'
+import { Admonition } from 'ui-patterns'
+import Link from 'next/link'
+import { AddonVariantId } from 'data/subscriptions/types'
+
+export function DiskManagementForm() {
+  const {
+    project,
+    // isLoading is used to avoud a useCheckPermissions() race condition
```

Comment:
the spell checker (do we still have that?) will probably error out here
```suggestion
    // isLoading is used to avoid a useCheckPermissions() race condition
```